src/plots/smith/smith.js

```diff
@@ -0,0 +1,1456 @@
+        ax.tick0 = rad2deg(ax.tick0);
+        ax.dtick = rad2deg(ax.dtick);
+    }
+
+    var _transFn = function(rad) {
+        return strTranslate(cx + radius * Math.cos(rad), cy - radius * Math.sin(rad));
+    };
+
+    var transFn = function(d) {
+        return _transFn(t2g(d));
+    };
+
+    var gridPathFn = function(d) {
+        var value = d.x;
+        var radius = _this.radius / d.x;
+
+        var arc = 2.0 * Math.atan(Math.abs(d.x));
+        var startAngle = value > 0.0 ? 1.5 * Math.PI - arc : Math.PI / 2.0;
+        var endAngle = value > 0.0 ? 1.5 * Math.PI : Math.PI / 2.0 + arc;
+
+        return Lib.pathArc(Math.abs(radius), startAngle, endAngle, cx + _this.radius, cy - radius);
+    };
+
+    var out = Axes.makeLabelFns(ax, 0);
+    var labelStandoff = out.labelStandoff;
+    var labelFns = {};
+
+    labelFns.xFn = function(d) {
+        var rad = t2g(d);
+        return Math.cos(rad) * labelStandoff;
+    };
+
+    labelFns.yFn = function(d) {
+        var rad = t2g(d);
+        var ff = Math.sin(rad) > 0 ? 0.2 : 1;
+        return -Math.sin(rad) * (labelStandoff + d.fontSize * ff) +
+            Math.abs(Math.cos(rad)) * (d.fontSize * MID_SHIFT);
+    };
+
+    labelFns.anchorFn = function(d) {
+        var rad = t2g(d);
+        var cos = Math.cos(rad);
+        return Math.abs(cos) < 0.1 ?
+            'middle' :
+            (cos > 0 ? 'start' : 'end');
+    };
+
+    labelFns.heightFn = function(d, a, h) {
+        var rad = t2g(d);
+        return -0.5 * (1 + Math.sin(rad)) * h;
+    };
+
+    var newTickLayout = strTickLayout(angularLayout);
+    if(_this.angularTickLayout !== newTickLayout) {
+        layers['angular-axis'].selectAll('.' + ax._id + 'tick').remove();
+        _this.angularTickLayout = newTickLayout;
+    }
+
+    var vals = Axes.calcTicks(ax);
+
+    // angle of polygon vertices in geometric radians (null means circles)
+    // TODO what to do when ax.period > ax._categories ??
+    var vangles;
+    if(smithLayout.gridshape === 'linear') {
+        vangles = vals.map(t2g);
+
+        // ax._vals should be always ordered, make them
+        // always turn counterclockwise for convenience here
+        if(Lib.angleDelta(vangles[0], vangles[1]) < 0) {
+            vangles = vangles.slice().reverse();
+        }
+    } else {
+        vangles = null;
+    }
+    _this.vangles = vangles;
+
+    // Use tickval filter for category axes instead of tweaking
+    // the range w.r.t sector, so that sectors that cross 360 can
+    // show all their ticks.
+    if(ax.type === 'category') {
+        vals = vals.filter(function(d) {
+            return Lib.isAngleInsideSector(t2g(d), _this.sectorInRad);
+        });
+    }
+
+    if(ax.visible) {
+        Axes.drawGrid(gd, ax, {
+            // vals: vals,
+            vals: [-5.0, -2.0, -1.0, -0.5, -0.2, 0.2, 0.5, 1.0, 2.0, 5.0].map(function(v) {
+                return {
+                    x: v,
+                };
+            }),
```

Comment:
This looks like a constant. Let's declare it once at the top.

Reply:
cb07c28f397c2b98f059b47fbacf5870039f15c0